packages/jupyter-widgets/src/manager/widgetLoader.ts

```diff
@@ -0,0 +1,89 @@
+import * as base from "@jupyter-widgets/base";
```

Comment:
That's a good point. I think it makes sense to have function singletons in the loader for now because we are opening up extensibility points for users to just be able to provide/override a part of it. So, a user could just provide a custom loader but still have nteract define all the dependencies for requireJS. Tying it into a class and ensuring that they are singletons that get passed around correctly is a little tricky for now. I think once we stabilize this implementation and build confidence in the community, it will probably make sense to define an interface and provide default implementations for functionality that people don't want to override. So, I wanted to start with a flexible option following the jupyter-widgets examples - https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/html-manager/src/libembed-amd.ts